### ISSUE TITLE
rpm leaks \x1f at argument boundaries in %global ... %(... %{?**})

### ISSUE THREAD

**Issue by mlschroe** [rpm-software-management/rpm#3145]
See https://bugzilla.redhat.com/show_bug.cgi?id=2284187

Reproducer taken from comment #4:
```
$ cat prinargv.py 
import sys
print(sys.argv)
$ rpm --eval '%define macro(e:) %{-e:%global toxenv %(%{__python3} -s ./prinargv.py %{?**})}'  --eval '%macro -e1 -e 2' --eval '%toxenv'


['./prinargv.py', '\x1f-e1\x1f', '\x1f-e\x1f', '\x1f2\x1f']

```

**Comment by mlschroe** [comment]
The fix is just one line, I'll open a pull request.

**Comment by pmatilai** [comment]
Ack, thanks a lot for looking into it!

**Comment by pmatilai** [comment]
Fixed by #3148
